clevertap-core/src/main/java/com/clevertap/android/sdk/CleverTapAPI.java

```diff
@@ -1089,6 +1093,47 @@ public CustomTemplateContext getActiveContextForTemplate(@NonNull String templat
         return coreState.getTemplatesManager().getActiveContextForTemplate(templateName);
     }
 
+    /**
+     * Sync all currently registered templates (through {@link #registerCustomInAppTemplates(TemplateProducer)} to the
+     * backend. Use this method to transfer template definitions from the SDK to the CT dashboard. This method can
+     * only be called from a debug build of the application and the current logged-in user must be marked as a
+     * test profile through the web dashboard.
+     */
+    public void syncRegisteredInAppTemplates() {
+        if (!isDevelopmentMode()) {
+            getConfigLogger().debug("CustomTemplates",
+                    "Your app is NOT in development mode, templates will not be synced");
+            return;
+        }
+
+        if (coreState == null) {
+            getConfigLogger().debug("CustomTemplates",
+                    "coreState is null, templates will cannot be synced");
+            return;
+        }
+
+        if (!(coreState.getNetworkManager() instanceof NetworkManager)) {
+            getConfigLogger().debug("CustomTemplates",
+                    "networkManager is not of correct type, templates will cannot be synced");
+            return;
+        }
+
+        if (coreState.getTemplatesManager() == null) {
+            getConfigLogger().debug("CustomTemplates",
+                    "templateManager is null, templates will cannot be synced");
+            return;
+        }
+
+        getCleverTapID(x -> {
```

Comment:
the listeners are not stored in a list. Potentially we might end up missing out on some other listener.
`public void syncVariables()` also has this method call. 
Commenting so you can investigate about this.

Reply:
Good call, the current implementation of getCleverTapID could have led to problems. I have changed the listeners to be a list.